telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
     failed_set[url] = output
     pickle.dump(failed_set, open('failed_urls', 'a'))
 
+def copy_wpr_to_benchmark():
+    """Moves wpr and .pc.wpr (all) files from local data/wpr_source to chromium
+
+    located: /src/tools/perf/page_sets/data/
+    """
+    local_path = os.path.join(os.getcwd(), 'data/wpr_source/*')
+    remote_data_path = os.path.join(CHROMIUM_SRC, 'tools/perf/page_sets/data/')
+    p = Popen('cp {0} {1}'.format(local_path, remote_data_path), shell=True)
+
+def prepare_benchmark(trial_number, num_urls):
+    """Simply checks to make sure the env is prepped before benchmarking
+
+    1. Ensures that ~/page_load_time/telemetry/temp/benchmark_results.db exists
+    2. Writes benchmarks to ~/src/tools/perf/benchmarks/page_cycler.py if they
+       do not already exist
+    NOTE: You may need to clear out the end of
+    ~/src/tools/perf/benchmarks/page_cycler.py for a clean start!
+    :param trial_number: int for the number of this trial run
+    :param num_urls: int of the number of urls to run through
+    """
+    # Ensure benchmark_results.db exists
+    telemetry_record_path = \
+    '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db'
+    if not os.path.isfile(telemetry_record_path):
+        # Create file
+        tmp = {trial_number: {}}
+        pickle.dump(tmp, open(telemetry_record_path, 'wb'))
+    try:
+        curr_pickle = pickle.load(open(telemetry_record_path, 'rb'))
+    except IOError:
+        # Raise exception or rewrite benchmark_results.db?
+        tmp = {trial_number: {}}
+        pickle.dump(tmp, open(telemetry_record_path, 'wb'))
+
+    # Write benchmarks to page_cycler.py
+    page_cycler_path = '/home/jamshed/src/tools/perf/benchmarks/page_cycler.py'
+    benchmark_template = ("@benchmark.Enabled('android')\n"
+                          "class PageCyclerUrl{0}(_PageCycler):\n"
+                          "    page_set = page_sets.url{0}PageSet\n\n")
+
+    with open(page_cycler_path, 'r+b') as f:
+        line = f.readline()
+        already_written = False
+        while (line):
+            if '### END OF URLS ###' in line:
```

Comment:
This would be deprecated by your comment above - just writing all of page_cycler.py instead of appending to it. This feels a bit hacky because it would over write future chromium updates, but it would ensure the benchmark is fully loaded every time.